pkg/option/config.go

```diff
@@ -2143,6 +2144,90 @@ func (c *DaemonConfig) Populate() {
 		c.ipv4NativeRoutingCIDR = cidr.MustParseCIDR(nativeCIDR)
 	}
 
+	// BPF map size options
+	// Any map size explicitly set via option will override the dynamic
+	// sizing.
+	c.CTMapEntriesGlobalTCP = viper.GetInt(CTMapEntriesGlobalTCPName)
+	c.CTMapEntriesGlobalAny = viper.GetInt(CTMapEntriesGlobalAnyName)
+	c.NATMapEntriesGlobal = viper.GetInt(NATMapEntriesGlobalName)
+	c.PolicyMapMaxEntries = viper.GetInt(PolicyMapEntriesName)
+
+	if viper.GetBool(MapEntriesGlobalDynamicSizeName) {
+		vms, err := mem.VirtualMemory()
+		if err != nil || vms == nil {
+			log.WithError(err).Fatal("Failed to get system memory")
+		}
+		log.Debugf("memory: %v total", vms.Total)
+
+		// Heuristic: use 1% of total memory for maps.
+		// Distribute relative to map default entries among the
+		// different maps. Cap each map size by the maximum. Map size
+		// provided by the user will override the calculated value and
+		// also the max. There will be check for maximum size later on
+		// in DaemonConfig.Validate()
+		memoryAvailableForMaps := int(vms.Total / 100)
```

Comment:
Should the percentage of memory used for maps be configurable? I guess we can also expose it later if there's demand.

Reply:
Good point. Initially I was afraid that this would give users a way to easily shot themselves in the foot with an OOM, but given that we cap the min/max of the map entries, I think it would be safe to allow a percentage here (along with sufficient documentation on how to use it, e.g. with some example calculations as suggested in https://github.com/cilium/cilium/pull/10780#discussion_r404199537). And given that we enable this by default as @tgraf suggested, we could convert the option to take a percentage instead of a boolean.

Reply:
That would be good added value for the users if we provide the information.

Reply:
Changed the command line option to take a ratio and and this is now used instead of the fixed value of 1%